src/components/Checkbox/index.tsx

```diff
@@ -208,7 +203,7 @@ const Checkbox = forwardRef(
     ref: ForwardedRef<HTMLInputElement>,
   ) => {
     const hasChildren = !!children
-    const { state, setState } = useCheckboxState({ state: checked })
+    const [state, setState] = useState<boolean | 'indeterminate'>(checked)
```

Comment:
Should we remove this behaviour ? This component doesn't need to handle its own state. The user should use checked prop and onChange to change it ?